### PR TITLE
Add FCM plugin support

### PR DESCRIPTION
**Pull request by rolandjitsu**

### CODE REVIEW

File: src/plugins/fcm.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    'iOS'
+  ]
+})
+export class FCM {
+  /**
+   * Get notified every time a new token is generated.
+   * @return {Observable<any>}
+   */
+  @Cordova({
+    callbackStyle: 'node',
```
Comment (ihadeed) [inline src/plugins/fcm.ts]: `callbackStyle: 'node'` is used incorrectly here. It's not needed.

`callbackStyle: 'node'` is used when the plugin expects a callback like this one:
```
function(err, result) { }
```

File: src/plugins/fcm.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  static unsubscribeFromTopic(topic: string): Promise<any> {
+    return;
+  }
+
+  /**
+   * Receive push notification data.
+   * @return {Observable<any>}
+   */
+  @Cordova({
+    callbackStyle: 'node',
```
Comment (ihadeed) [inline src/plugins/fcm.ts]: Remove `callbackStyle: 'node'` from here too. See comment above.

File: src/plugins/fcm.ts (hunk 1 above)
Comment (rolandjitsu) [inline src/plugins/fcm.ts]: @ihadeed ok, I thought that [this](https://github.com/fechanique/cordova-plugin-fcm/blob/master/www/FCMPlugin.js#L21) qualifies for `callbackStyle: 'node'`.

**Comment by jsayol** [comment]
Just curious, is this plugin necessary when [the current Firebase plugin](https://ionicframework.com/docs/v2/native/firebase/) already offers all of FCM features and more?

**Comment by ihadeed** [comment]
@jsayol yeah you're right that plugin has everything. I don't think this is needed.

**Pull request closed without merging**